tests/all.rs

```diff
@@ -1149,3 +1149,40 @@ fn append_long_multibyte() {
         name.pop();
     }
 }
+
+#[test]
+#[cfg(unix)]
+fn tar_directory_containing_special_files() {
+    use std::env;
+    use std::ffi::CString;
+
+    let td = t!(TempBuilder::new().prefix("tar-rs").tempdir());
+    let fifo = td.path().join("fifo");
+
+    unsafe {
+        let fifo_path = t!(CString::new(fifo.to_str().unwrap()));
+        let ret = libc::mknod(fifo_path.as_ptr(), libc::S_IFIFO | 0o644, 0);
+        if ret != 0 {
+            libc::perror(fifo_path.as_ptr());
+            panic!("Failed to create a FIFO file");
+        }
+    }
+
+    t!(env::set_current_dir(td.path()));
+    let mut ar = Builder::new(Vec::new());
+    // append_path has a different logic for processing files, so we need to test it as well
+    t!(ar.append_path("fifo"));
+    t!(ar.append_dir_all("special", td.path()));
+    // unfortunately, macOS CI can't handle `set_current_dir` correctly
+    if !cfg!(target_os = "linux") {
```

Comment:
Could the `#[cfg]` on the test be updated to `target_os = "linux"`?

Reply:
Addressed in 7b5fec1f91e961b2fafe6a897b36fd9d06366978